### PR TITLE
[1LP][RFR] Check that Nuage events are emitted and received

### PR DESCRIPTION
**Pull request by miha-plesko**
With this commit we add a test to verify that events are being recieved by MIQ.

### CODE REVIEW

**Comment by mshriver** [comment]
This needs requirements update to wrapanapi 3.0.24, where `delete_enterprise` is supported.

**Comment by miha-plesko** [comment]
Forgot about it, thanks.

**Comment by miha-plesko** [comment]
Awesome the related framework PR https://github.com/ManageIQ/integration_tests/pull/7722 has been merged so I think we're good with this one as well.

**Comment by miha-plesko** [comment]
Is there something else to be improved on this one? @psav or @quarckster maybe?

File: cfme/tests/networks/nuage/test_events_are_triggered.py [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+from cfme.networks.provider.nuage import NuageProvider
+from cfme.utils.log import logger
+
+
+pytestmark = [
+    pytest.mark.provider([NuageProvider])
+]
+
+
+@pytest.yield_fixture
```
Comment (izapolsk) [inline cfme/tests/networks/nuage/test_events_are_triggered.py]: yield_fixture is deprecated.  its functionality has been moved to pytest.fixture. please use it instead.

File: cfme/tests/networks/nuage/test_events_are_triggered.py [hunk 2]
```diff
@@ -0,0 +1,121 @@
+# -*- coding: utf-8 -*-
+"""This module tests Nuage EMS events."""
+import pytest
+from wrapanapi.utils.random import random_name
```
Comment (izapolsk) [inline cfme/tests/networks/nuage/test_events_are_triggered.py]: please sort imports according to our guidelines -  https://cfme-tests.readthedocs.io/en/master/guides/dev_guide.html

File: cfme/tests/networks/nuage/test_events_are_triggered.py [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+    box['template'] = enterprise.create_child(
+        nuage.vspk.NUDomainTemplate(name=random_name()))[0]
+    box['domain'] = enterprise.create_child(
+        nuage.vspk.NUDomain(name=random_name(), template_id=box['template'].id))[0]
+    box['zone'] = box['domain'].create_child(
+        nuage.vspk.NUZone(name=random_name()))[0]
+    box['subnet'] = box['zone'].create_child(
+        nuage.vspk.NUSubnet(
+            name=random_name(),
+            address='192.168.0.0',
```
Comment (izapolsk) [inline cfme/tests/networks/nuage/test_events_are_triggered.py]: Q: shouldn't such data be stored in cfme_data yaml ?
as far as I understand it is provider related.

File: cfme/tests/networks/nuage/test_events_are_triggered.py [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+    Nuage server, event of type 'nuage_subnet_create' should be emitted.
+
+    Prerequisities:
+        * A network provider that is set up and listening to events.
+
+    Steps:
+        * Deploy some entities outside of MIQ (directly in the provider)
+        * Verify that expected events were captured by MIQ
+
+    Important: register_event fixture must be listed before with_nuage_sandbox fixture because we
```
Comment (izapolsk) [inline cfme/tests/networks/nuage/test_events_are_triggered.py]: :+1:

**Pull request merged**